code-challenges.test.js

```diff
@@ -18,54 +18,165 @@
+
+
+//??? i tried a few different approches but ended up digging myself a deeper hole with this jest test. 
+//the function works just not the test. 
+
+// describe("when shuffleColors is called", () =>{
+//     it("if argument is colors1, return slice array",() =>{
+//         const argument = "colors1";
+//         const actualResults = shuffleColors(argument)
+//         expect(actualResults).toEqual(shuffleColors(colors1).sort())
+//     })
+//     it("if argument is colors2, return slice shuffleColors(colors1).sort()",() =>{
+//         const argument = colorsArray;
+//         const actualResults = shuffleColors(argument)
+//         expect(actualResults).toEqual(colorsArray.sort())
+//     })
+// })
+//for colors1 expect ["yellow", "blue", "pink", "green"]or something else with only 4 index 1->4 index 0 removed
+
+
+// b) Create the function that makes the test pass.
+
+//create an function that will take in an array as the argumnent
+
+const shuffleColors = (array) =>{
+    //filter the array will slice the index of 0
+    var colorsArray = array.slice(1,)
+    //filter the array will slice the index of 0
+//push (Math.floor *Math.random) for index new array to newArray
+    var i = colorsArray.length, j, temp;
+    while(--i >0) {
+        j= Math.floor(Math.random() * (i+1));
+        temp= colorsArray[j];
+        colorsArray[j] = colorsArray[i];
+        colorsArray[i] = temp;
+    }
+
+return colorsArray
+
+}
+
+//
+
 var colors1 = ["purple", "blue", "green", "yellow", "pink"]
 // Expected output example (can be a different order): ["yellow", "blue", "pink", "green"]
 var colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
 // Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
 
 
+console.log(shuffleColors(colors1))
+console.log(shuffleColors(colors2))
 
 
 
-// b) Create the function that makes the test pass.
-
-
 
 
 
 // --------------------2) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
 
 // a) Create a test with expect statements for each of the variables provided.
+//setting up test that will describe when cubedSum gets called 
+//it will take the argument and *** by itself and add each new number in that array to get a total. 
+describe("when sumUp is called", () =>{
+    it("if argument is cubeAndSum1, return 99",() =>{
+        const argument = cubeAndSum1;
+        const actualResults = sumUp(argument)
+        expect(actualResults).toEqual(99)
+    })
+    it("if argument is cubeAndSum2, return 1125",() =>{
+        const argument = cubeAndSum2;
+        const actualResults = sumUp(argument)
+        expect(actualResults).toEqual(1125)
+    })
+})
+
+
+// b) Create the function that makes the test pass.
+//setting up function that takes in an array of numbers as the argument
+
+
+const sumUp = (array) => {
+    let cubedArray =[]
+    for(let i= 0; i < array.length ; i++){
```

Comment:
looks like this would be a good opportunity to use `.map()` since cubedArray is expected to be the same length as the array provided as an argument